src/lcm/create_params.py

```diff
@@ -1,138 +1,176 @@
 """Create a parameters for a model specification."""
 
 import inspect
+from typing import Any
 
-import numpy as np
+import jax.numpy as jnp
+import pandas as pd
+from jax import Array
 
 
-def create_params(user_model, variable_info, grids):
-    """Get parameters from a model specification.
+def create_params_template(
+    model_spec: dict[str, Any],
+    variable_info: pd.DataFrame,
+    grids: dict[str, Array],
+    default_params: dict[str, float] | None = None,
+) -> dict[str, dict[str, Any] | float]:
+    """Create parameter template from a model specification.
 
     Args:
-        user_model (dict): A model specification. Has keys
+        model_spec (dict): A model specification provided by the user. Has keys
             - "functions": A dictionary of functions used in the model.
             - "choices": A dictionary of choice variables.
             - "states": A dictionary of state variables.
             - "n_periods": Number of periods in the model (int).
             - "shocks": A dictionary of shock variables (optional).
         variable_info (pd.DataFrame): A dataframe with information about the variables.
-        grids (dict): A dictionary of grids.
+        grids (dict): A dictionary of grids retrieved from model_spec.
```

Comment:
```suggestion
        grids (dict): A dictionary of grids consistent with model_spec.
```
Open to other suggestions, but the original version triggered me to think about whether we need it at all.